hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/test/StreamSourceLong.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.pipeline.test;
+
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.AppendableTraverser;
+import com.hazelcast.jet.core.EventTimePolicy;
+import com.hazelcast.jet.core.Watermark;
+
+import javax.annotation.Nonnull;
+
+import static com.hazelcast.jet.impl.JetEvent.jetEvent;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Represents a StreamSource containing Long values which can be used
+ * for performance pipeline testing and development.
+ *
+ * @since 4.3
+ */
+public class StreamSourceLong extends AbstractProcessor {
+
+    private static final long SOURCE_THROUGHPUT_REPORTING_PERIOD_SECONDS = 10;
+    private static final long SIMPLE_TIME_SPAN_MILLIS = HOURS.toMillis(3);
+
+    private static final long REPORT_PERIOD_NANOS = SECONDS.toNanos(SOURCE_THROUGHPUT_REPORTING_PERIOD_SECONDS);
+    private static final long HICCUP_REPORT_THRESHOLD_MILLIS = 10;
+    private final long nanoTimeMillisToCurrentTimeMillis = determineTimeOffset();
+    private final long startTime;
+    private final long itemsPerSecond;
+    private final boolean isReportingThroughput;
+    private final long wmGranularity;
+    private final long wmOffset;
+    private long globalProcessorIndex;
+    private long totalParallelism;
+    private long emitPeriod;
+
+    private final AppendableTraverser<Object> traverser = new AppendableTraverser<>(2);
+    private long emitSchedule;
+    private long lastReport;
+    private long counterAtLastReport;
+    private long lastCallNanos;
+    private long counter;
+    private long lastEmittedWm;
+    private long nowNanos;
+
+    /**
+     * Creates a stream source for Long values that is supposed to be used for performance testing purposes.
```

Comment:
Javadoc must be wrapped so that line length is 72 chars (ignoring initial indentation or `*`)

Reply:
Done w/ 8fcf10859c9274eca0f80fe5f7b45bac375000dc